### ISSUE TITLE
Set quiet period on Jenkins builds.

### ISSUE THREAD

**Issue by hhellyer** [eclipse/codewind#2408]
**Description of the enhancement:**

When we merge a PR this triggers not only a new build of `master` but also a new build for every outstanding branch.
As a committer my usual workflow is to allocate some time to review PRs which results in several being merged fairly closely to each other.
This triggers a lot of builds `(merged_pr_count) * (1 + outstanding_pr_count)` - often our PR queue is > 10 so it's easy to trigger dozens of builds.
This results in a lot of load on the build system. It also results in the build triggered by the first merge invalidating the build status of other PRs that are awaiting review and all but one master build will be out of date before it completes.

**Proposed solution:** 
Enable the "Quiet Period" setting on Jenkins: https://ci.eclipse.org/codewind/job/Codewind/job/codewind/job/master/configure

We used a similar setting previously on travis, this prevents a build from starting immediately a merge occurs.

Additionally if committers are aware of this behaviour they can review and approve several PR's so they are ready to merge then do the actual merges together to limit the number of builds they trigger.

I would suggest setting a delay of 5 minutes (300 seconds) to give a human being time to merge multiple PRs. This will delay master builds slightly but hopefully by prevent the Jenkins from being swamped will actually speed up build responsiveness overall.

Docs:

> When this option is non-zero, newly triggered builds of this project will be added to the queue, but Jenkins will wait for the specified period of time (in seconds) before actually starting the build.
> For example, if your builds take a long time to execute, you may want to prevent multiple source control commits that are made at approximately the same time from triggering multiple builds. Enabling the quiet period would prevent a build from being started as soon as Jenkins discovers the first commit; this would give the developer the chance to push more commits which would be included in the build when it starts. This reduces the size of the queue, meaning that developers would get feedback faster for their series of commits, and the load on the Jenkins system would be reduced.
> 
> If a new build of this project is triggered while a build is already sitting in the queue, waiting for its quiet period to end, the quiet period will not be reset. The newly triggered build will not be added to the queue, unless this project is parameterized and the build has different parameters than the build already in the queue.
>

**Comment by hhellyer** [comment]
I've raised https://github.com/eclipse/codewind/pull/2409 and asked @josephca to review.

**Comment by hhellyer** [comment]
If this is successful for codewind we should look at doing it on some of the other respositories like codewind-installer.

**Comment by josephca** [comment]
In addition to adding the quiet period, I increased the repository scan intervals from 5 mins to 10 mins (from Jenkins configuration) which would help in the same way.

**Comment by josephca** [comment]
@hhellyer  Second thought - I don't think the change would help much in reducing the number of builds. Because, currently, the repository scan is scheduled to run every 5 mins and if it detects any changes in branches and PRs, it will kick off builds anyways. Adding a quiet period is simply making the build slower. 
Instead, I increased the repository scan intervals from 5 to 10 mins from Jenkins configuration which would help a bit to avoid too frequent rebuild. However, it also depends on when the merge is done. i.e. right after the scheduled repository scan vs. right before the scan. 

I merged the change as I thought it would help initially.  If it doesn't help, we may need to revert it.
